airflow/providers/ssh/hooks/ssh.py

```diff
@@ -491,9 +491,12 @@ def exec_ssh_client_command(
         if stdout_buffer_length > 0:
             agg_stdout += stdout.channel.recv(stdout_buffer_length)
 
+        timedout = False
+
         # read from both stdout and stderr
         while not channel.closed or channel.recv_ready() or channel.recv_stderr_ready():
             readq, _, _ = select([channel], [], [], timeout)
+            timedout = len(readq) == 0
```

Comment:
I think you need to check if **all** three results are empty to be sure a timeout has occurred. Useful link: https://stackoverflow.com/a/15195460/1055702

Reply:
I'm not sure - from `select` doc
```
the return value is a tuple of three lists corresponding to the first three arguments; 
each contains the subset of the corresponding file descriptors that are ready.
```
and we pass empty list to 2nd and 3rd args.

Reply:
I don't have a deep enough understanding of the select api and ssh to know for sure, but it seems harmless to check the others.

If you're positive that this is correct then I'm happy to commit and approve.

Reply:
Since we're passing empty list for the 2nd and 3rd arg, i dont see how the other 2 cannot be empty. It makes sense to pass empty file descriptors since the 2nd arg is for writing and the 3rd for exception condition.

Reply:
Okay, sounds good, approving :)

Reply:
what's the next step to get this merged?

Reply:
> what's the next step to get this merged?

There are some static checks that are failing which need fixing before it can be merged. You can enable pre-commit to run static checks on your code before submitting a PR update ([readme](https://github.com/apache/airflow/blob/main/STATIC_CODE_CHECKS.rst#installing-pre-commit-hooks)) 

Then someone with committer status will need to review/approve/merge the change :) (CC: @eladkal @potiuk)

Reply:
got it - i think i fixed it - re-running now

Reply:
Thank you @potiuk anything else we need to merge it?